tools/cbmc/proofs/DHCPProcess/DHCPProcess_harness.c

```diff
@@ -0,0 +1,61 @@
+#include <stdint.h>
+
+/* FreeRTOS includes. */
+#include "FreeRTOS.h"
+#include "task.h"
+
+/* FreeRTOS+TCP includes. */
+#include "FreeRTOS_IP.h"
+#include "FreeRTOS_IP_Private.h"
+#include "FreeRTOS_DHCP.h"
+
+// CBMC automatically unwinds strlen on a fixed string
+const char * pcApplicationHostnameHook(void) {
+  return "hostname";
+}
+
+// This stub allows us to overcome the unwinding error obtained
+// in the do-while loop within function prvCreatePartDHCPMessage.
+// The behaviour is similar to the original function, but failed allocations
+// are not considered here (this is a safe assumption that avoids the error)
+void *FreeRTOS_GetUDPPayloadBuffer( size_t xRequestedSizeBytes, TickType_t xBlockTimeTicks )
+{
+  NetworkBufferDescriptor_t pxNetworkBuffer;
+  void *pvReturn;
+
+  pxNetworkBuffer.xDataLength = ipUDP_PAYLOAD_OFFSET_IPv4 + xRequestedSizeBytes;
+  pxNetworkBuffer.pucEthernetBuffer = malloc( pxNetworkBuffer.xDataLength );
+  pvReturn = (void *) &( pxNetworkBuffer.pucEthernetBuffer[ ipUDP_PAYLOAD_OFFSET_IPv4 ] );
+  return pvReturn;
+}
+
+// In this stub we allocate a buffer within the specified range
+// Note that the values for BUFFER_SIZE and prvProcessDHCPReplies.0
+// are correlated, and their current values have been adjusted in order
+// to obtain maximum coverage in the shortest amount of time
+int32_t FreeRTOS_recvfrom(
+  Socket_t xSocket, void *pvBuffer, size_t xBufferLength,
+  BaseType_t xFlags, struct freertos_sockaddr *pxSourceAddress,
+  socklen_t *pxSourceAddressLength )
+{
+  __CPROVER_assert(xFlags & FREERTOS_ZERO_COPY, "I can only do ZERO_COPY");
+
+  size_t buffer_size;
```

Comment:
According to the naming conventions: xBufferSize

https://www.freertos.org/FreeRTOS-Coding-Standard-and-Style-Guide.html